diff --git a/components/SidebarSublist.astro b/components/SidebarSublist.astro
index 232525016ee16f1ad1fe8f06a330bec4962ad63f..e97a91fb2b97baa133ccb1eb885f807f5471b1a8 100644
--- a/components/SidebarSublist.astro
+++ b/components/SidebarSublist.astro
@@ -19,7 +19,7 @@ const { sublist, nested } = Astro.props;
 					<a
 						href={entry.href}
 						aria-current={entry.isCurrent && 'page'}
-						class:list={[{ large: !nested }, entry.attrs.class]}
+						class:list={[{ large: !nested || sublist.some(e => e.type != 'link') }, entry.attrs.class]}
 						{...entry.attrs}
 					>
 						<span>{entry.label}</span>
